src/api/Flooq.Test/Controllers/DataFlowControllerTest.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Flooq.Api.Controllers;
+using Flooq.Api.Models;
+using Flooq.Api.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace Flooq.Test.Controllers;
+
+[TestClass]
+public class DataFlowControllerTest
+{
+  private readonly Mock<IDataFlowService> _serviceMock = new Mock<IDataFlowService>();
+  private readonly DataFlow _dataFlow = new DataFlow
+  {
+    Id = Guid.NewGuid(),
+    Name = "Demo Flow",
+    Status = "Active",
+    LastEdited = DateTime.Now,
+    Definition = "{}"
+  };
+    
+  
+  [TestMethod]
+  public void CanCreateDataFlowController()
+  {
+    var mockDataFlowService = _serviceMock.Object;
+    var dataFlowController = new DataFlowController(mockDataFlowService);
+    Assert.IsNotNull(dataFlowController);
+  }
+
+  [TestMethod]
+  public void CanGetDataFlows()
+  {
+    var dataFlows = new List<DataFlow>();
+    dataFlows.Add(_dataFlow);
+    
+    var actionResult = new ActionResult<IEnumerable<DataFlow>>(dataFlows);
+    _serviceMock.Setup(service => service.GetDataFlows()).ReturnsAsync(actionResult);
+
+    var dataFlowController = new DataFlowController(_serviceMock.Object);
+    
+    Assert.AreSame(actionResult, dataFlowController.GetDataFlows().Result);
+  }
+
+  [TestMethod]
+  public void CanGetDataFlow()
+  {
+    _serviceMock.Setup(service => service.GetDataFlow(_dataFlow.Id)).ReturnsAsync(_dataFlow);
+
+    var dataFlowController = new DataFlowController(_serviceMock.Object);
+
+    Assert.AreSame(_dataFlow, dataFlowController.GetDataFlow(_dataFlow.Id).Result.Value);
+  }
+
+  [TestMethod]
+  public async Task Get_ReturnsNotFoundIfThereIsNoMatchingDataFlow()
+  {
+    _serviceMock.Setup(service => service.GetDataFlow(_dataFlow.Id)).ReturnsAsync(new ActionResult<DataFlow>(_dataFlow));
+    
+    var dataFlowController = new DataFlowController(_serviceMock.Object);
+
+    ActionResult<DataFlow> result = await dataFlowController.GetDataFlow(Guid.NewGuid());
+    
+    Assert.IsNotNull(result.Result);
+    Assert.AreEqual(new NotFoundResult().ToString(), result.Result.ToString());
```

Comment:
Instead of the Assert.AreEqual, could you not write something like `Assert.IsInstanceOfType(result.Result, typeof(NotFoundResult))`?